tcms/xmlrpc/api/telemetry.py

```diff
@@ -0,0 +1,64 @@
+from modernrpc.core import rpc_method
+
+from tcms.management.models import Product
+from tcms.testcases.models import TestCase
+from tcms.testplans.models import TestPlan
+
+
+@rpc_method(name='Telemetry.get_data')
+def get_data(request):
+
+    test_plan_id = request.get('test_plan_id')
+    if not test_plan_id:
+        raise Exception('Required test_plan_id')
+
+    test_plan = TestPlan.objects.get(plan_id=test_plan_id)
+
+    test_cases = TestCase.objects.filter(plan=test_plan)
```

Comment:
Filter test cases based on `query` above. If user needs to filter by plan they can supply the appropriate field lookups. This will make the input of this method the same as `TestCase.filter` and then we can work to reuse the filtering template and JavaScript from test case search page.